src/fdc/DiskImageUtils.cc

```diff
@@ -189,34 +192,42 @@ static SetBootSectorResult setBootSector(
 	unsigned nbRootDirSectors = nbDirEntry / 16;
 	unsigned rootDirStart = nbReservedSectors + nbFats * nbSectorsPerFat;
 	unsigned firstDataSector = rootDirStart + nbRootDirSectors;
-	return {nbReservedSectors, firstDataSector, descriptor};
+	return {nbFats, nbSectorsPerFat, nbReservedSectors, rootDirStart, firstDataSector, descriptor};
 }
 
 void format(SectorAccessibleDisk& disk, bool dos1)
 {
 	// first create a boot sector for given partition size
 	size_t nbSectors = disk.getNbSectors();
 	SectorBuffer buf;
-	auto [firstFATSector, firstDataSector, descriptor] =
+	auto [fatCount, sectorsPerFat, fatStart, rootDirStart, dataStart, descriptor] =
 		setBootSector(buf.bootSector, nbSectors, dos1);
 	disk.writeSector(0, buf);
 
 	// write empty FAT and directory sectors
 	ranges::fill(buf.raw, 0);
-	for (auto i : xrange(firstFATSector + 1, firstDataSector)) {
+	for (auto i : xrange(1u, sectorsPerFat)) {
+		for (auto fat : xrange(fatCount)) {
+			disk.writeSector(i + fatStart + fat * sectorsPerFat, buf);
+		}
+	}
```

Comment:
Suggestion, I haven't fully worked it out. Might be better or might not be.

You can swap the order of the inner and outer loop. Then the disk is accessed in a more linear pattern. Speed is not really important here, but it also allows to integrate the loop below that writes the first FAT sector.

Or maybe not because that requires changing the content of `buf` more often. And that might be more ugly.

Reply:
Yes, although I thought it would look nicer to have a single loop with an if (i == 0) in it, I didn’t do it for the latter reason.